core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -63,7 +66,10 @@ private TableIdentifier canonicalizeIdentifier(TableIdentifier tableIdentifier)
 
   @Override
   public Table loadTable(TableIdentifier ident) {
-    return tableCache.get(canonicalizeIdentifier(ident), catalog::loadTable);
+    Table result = tableCache.get(canonicalizeIdentifier(ident), catalog::loadTable);
+    LOG.info("Table loaded by catalog: {}", result);
```

Comment:
I think this is misleading. What we want to know is whether a cached version of the table was used or whether the catalog loaded the table. The catalog should log when it loads a table, so this just needs to log when there was a cache hit.

Reply:
I was thinking around the same lines, but I did not find a way to differentiate between the cache hit, and the cache miss. 
Shall we just remove the log, or you have some idea how to check for cache hit?
Removed the log for the time being.

Reply:
Well, a cache hit could just be the absence of a log message. But you could also inject an atomic boolean in the load table call that is passed to the cache. If it gets called, it sets the boolean to true. Then the log only triggers if it was not called.

Reply:
Ok, then cache hit will be marked by the absence of the log message.